spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -813,7 +876,15 @@ private void handleAck(long deliveryTag, boolean channelLocallyTransacted) throw
 							&& TransactionSynchronizationManager.getResource(this.connectionFactory) == null);
 			try {
 				if (this.ackRequired) {
-					if (!isChannelTransacted() || isLocallyTransacted) {
+					if (this.messagesPerAck > 0) {
```

Comment:
I think this one must be `> 1` as well. 
Literally the default `0` is fully equal to `1` - and we perform `basicAck()` on each message.